rfcs/20230704-dynamism-101.md

```diff
@@ -0,0 +1,825 @@
+# RFC: Dynamism 101
+
+Status: Under review<br/>
+Initial version: 7/4/2023<br/>
+Last updated: 7/4/2023<br/>
+Discussion thread: [openxla-discuss](https://groups.google.com/a/openxla.org/g/openxla-discuss/c/HJRvFBum65k/m/7QtJxgB9AQAJ).
+
+## Summary
+
+This RFC aims to leverage existing support for dynamism in the StableHLO
+dialect, discuss improvements to the existing design and then formalize the
+improved design in the StableHLO opset. To that end, this document proposes
+the following:
+
+<!-- markdownlint-disable line-length -->
+* [(P1) Move TensorFlow-specific operations out of StableHLO](#p1).
+* [(P2) Ratify the existing convention for shape mismatches constituting undefined behavior](#p2).
+* [(P3) Ratify the existing convention for relaxed constraints already implemented in the StableHLO dialect](#p3).
+* [(P4) Enable dynamism for all size-related program elements but keep all axis-related program elements static](#p4).
+* [(P5) Unify static and dynamic versions of StableHLO operations](#p5).
+* [(P6) Represent shape computations as StableHLO operations working on 0-dimensional tensors](#p6).
+<!-- markdownlint-enable line-length -->
+
+These proposals address a considerable chunk of community feedback on the
+existing design, but some feedback is deliberately left out of scope for this
+RFC to enable incremental progress while areas which require additional
+alignment are developed in parallel. See sections "Community feedback" and
+"Out of scope" for details.
```

Comment:
Could you link "community feedback" / "out of scope" similar to how you linked proposals?

Reply:
Done.